### PR TITLE
build: rust compilation support

### PR DESCRIPTION
**Pull request by jjohannes**
**Description**:
Integrate Gradle and Rust/Cargo:

- [x] Gradle tasks for Rust compilation
- [x] Install Rust toolchain via GitHub Actions
- [x] Build the `linux-x86-64` target
- [x] Enhance Rust toolchain setup to support cross-compilation for all desired targets
   - Windows cross-compilation using `lld` based on https://jake-shadle.github.io/xwin/ 

**Related issue(s)**:

https://github.com/hashgraph/hedera-services/issues/14100

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by mxtartaglia-sl — DISMISSED** [review]
This is great, Jendrik! With this implementation, I can stop uploading the binaries for unit test to pass.
The only comment I would make is that maybe it is worth having the targets in some sort of enum, but I didn't investigate whether that is possible.

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by mxtartaglia-sl — DISMISSED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by mxtartaglia-sl — DISMISSED** [review]

**Comment by jjohannes** [comment]
Once this is in, we could extend the pipeline to run all tests on runners with different operating systems. The built libraries are pulled from the Gradle cache. I have already tested that on a branch (#82) and [it works](https://github.com/hashgraph/hedera-cryptography/actions/runs/11368312816?pr=82). 

Discovered this issue: https://github.com/hashgraph/hedera-cryptography/issues/81

Prepared follow up PR (for once this is merged): #82

File: gradle/plugins/src/main/kotlin/com.hedera.gradle.feature.rust.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.hedera.gradle.services.TaskLockService
+import com.hedera.gradle.tasks.CargoBuildTask
+
+plugins { id("java") }
+
+val cargo = project.extensions.create<CargoExtension>("cargo")
+
+cargo.targets(aarch64Darwin, aarch64Linux, x86Darwin, x86Linux, x86Windows)
+
+// Cargo might do installation work, do not run in parallel:
```
Comment (rbarker-dev) [inline gradle/plugins/src/main/kotlin/com.hedera.gradle.feature.rust.gradle.kts:28]: why not run in parallel?

**Review by rbarker-dev — APPROVED** [review]

File: README.md
Comment (rbarker-dev) [inline README.md:1]: To get `./gradlew build` to succeed I also needed to execute the following command: `rustup target add x86_64-pc-windows-msvc`

Comment (rbarker-dev) [reply README.md:1]: I think we should put that in as a requirement for building on mac.

**Review by rbarker-dev — CHANGES_REQUESTED** [review]

File: README.md
Comment (jjohannes) [reply README.md:1]: Good catch! Added it to the Readme.

File: gradle/plugins/src/main/kotlin/com.hedera.gradle.feature.rust.gradle.kts (hunk 1 above)
Comment (jjohannes) [reply gradle/plugins/src/main/kotlin/com.hedera.gradle.feature.rust.gradle.kts:28]: Cargo/rust is doing installation/preparation work before compiling our code. If we run this in parallel, we start `cargo` multiple times in parallel. It then tries to do the same things multiple times in parallel and fails with files access issues (it's apparently not designed to run in parallel).
I tried to isolate this into [a share build service](https://docs.gradle.org/current/userguide/build_services.html), but did not succeed as this is triggered automatically when you start the (cross) compilation through cargo. We would somehow need to make sure that everything is prepared before the actual compilation starts. But I did not get it working (maybe cargo does support that at all).

**Review by rbarker-dev — APPROVED** [review]
LGTM

**Review by mxtartaglia-sl — APPROVED** [review]

**Comment by mxtartaglia-sl** [comment]
Tested locally and working. Both in Linux and in mac.

**Pull request merged**